src/app/services/dataset/dataset_processor_service.py

```diff
@@ -190,3 +195,15 @@ def _create_dataset_metadata_item_with_id(
         metadata_collection_item["dataset_id"] = dataset_metadata_snapshot.id
 
         return metadata_collection_item
+
+    def _extract_ruref_from_unit_data(
```

Comment:
I would slightly rename this to _extract_rurefs_from_unit_data_collection, the name right now makes it seem like it extracts a single ruref from some unit data